roles/common/files/qos.conf

```diff
@@ -0,0 +1,122 @@
+<IfModule qos_module>
+
+  ### Docs
+
+  ## This file is MOSTLY a set of limits that are applied to specific requests.  The requests are tagged with
+  #  counters like "ForumPage_PerIP_Short".  We use "Page" to mean an actual page that the user requests, rather than
+  #  a JS/CSS resource or an image.  We have separate _Short, _Med, and _Long counters so that we can have high
+  #  "burst" limits, which go down to more sustainable rates over longer time-frames.
+
+  ## Official manual for mod_qos:
+  #  https://mod-qos.sourceforge.net/
+  
+  ## ...and a blog post with a simple usage example, and some helpful comments:
+  #  https://ionica.ca/2021/08/24/mitigating-denial-of-service-attacks-with-mod_qos/
+
+  ## There's a "VIP" feature we might consider using in future.  As-is, our applications don't offer a reliable way
+  #  for mod_qos to check for "VIP" status, so we work from the opposite direction, giving generous(?) limits to
+  #  everyone, and then tightening things down where we can.
+  #  https://mod-qos.sourceforge.net/#privilegedusers
+  ### /Docs
+
+
+  ### Housekeeping
+
+  ## Return this status code when blocking a request
+  QS_ErrorResponseCode		429
+
+  ## Save logs of request rates, response times, and response status codes over time.
+  #QSLog "|/usr/bin/qslog -o /var/log/apache2/qos.log.csv"
+
+  ## Track stats for this number of clients.  It seems that less active clients will "roll off" the list over time.
+  #  Default is 50000.
+  #QS_ClientEntries		50000
+  ### /Housekeeping
+
+
+  ### TCP stream limits - close extra connections when we're busy
+
+  ## Defines the maximum allowed number of concurrent TCP connections for this server.
+  #  I'm not sure how this differs from MaxRequestWorkers (in mods-enabled/mpm_event.conf) which is set from Ansible
+  #QS_SrvMaxConn			300
```

Comment:
I'm worried about this, because while MaxRequestWorkers will limit how many requests can be _serviced_ at the same time, incoming requests are still queued up (based on my understanding of https://httpd.apache.org/docs/2.4/mod/mpm_common.html#maxrequestworkers). This would presumably immediately return a 429 to anything above the 300 concurrent limit. Which... is maybe what we want?